code/obj/item/implant.dm

```diff
@@ -890,6 +896,366 @@ ABSTRACT_TYPE(/obj/item/implant/revenge)
+						H.drop_item()
+					else
+						fail_reason = MARIONETTE_IMPLANT_ERROR_INVALID
+					src.adjust_heat(60)
+				else
+					fail_reason = MARIONETTE_IMPLANT_ERROR_DEAD_TARGET
+			if ("use", "activate")
+				if (!isdead(src.owner))
+					var/obj/item/I = src.owner.equipped()
+					if (istype(I))
+						logTheThing(LOG_COMBAT, src.owner, "was forced to activate \the [I] by \a [src] at [log_loc(src.owner)] (caused by [constructTarget(signal.author, "combat")] at [log_loc(signal.author)]).")
+						boutput(src.owner, SPAN_ALERT("Your hand involuntarily jerks."))
+						src.owner.click(I, list())
+					else
+						fail_reason = MARIONETTE_IMPLANT_ERROR_INVALID
+					src.adjust_heat(35)
+				else
+					fail_reason = MARIONETTE_IMPLANT_ERROR_DEAD_TARGET
+			else
+				fail_reason = MARIONETTE_IMPLANT_ERROR_INVALID
+		send_activation_reply(signal.data["sender"], fail_reason)
+
+	proc/send_activation_reply(sender_address, fail_reason)
+		var/datum/signal/activation_signal = get_free_signal()
+		activation_signal.source = src
+		activation_signal.data["device"] = "IMP_MARIONETTE"
+		activation_signal.data["sender"] = src.net_id
+		activation_signal.data["address_1"] = sender_address
+		activation_signal.data["command"] = "activate"
+		if (fail_reason)
+			activation_signal.data["stack"] = fail_reason
+		SEND_SIGNAL(src, COMSIG_MOVABLE_POST_RADIO_PACKET, activation_signal)
+
+	proc/adjust_heat(to_heat)
+		src.heat = max(0, src.heat + to_heat)
+		if (src.heat > src.heat_danger_zone && prob(20) && to_heat > 0)
+			SPAWN (0.25 SECONDS) // Give the implant time to send the activation reply
+				src.burn_out()
+
+	proc/burn_out()
+		if (ismob(src.owner))
+			logTheThing(LOG_COMBAT, src.owner, "had their [src.name] burn out and become useless.")
+			boutput(src.owner, SPAN_ALERT("You feel a painful burning, like there's a something hot inside your body."))
+			src.owner.TakeDamage("All", burn = 7, damage_type = DAMAGE_BURN)
+		if (src.linked_address)
+			var/datum/signal/burnout_signal = get_free_signal()
+			burnout_signal.source = src
+			burnout_signal.data["device"] = "IMP_MARIONETTE"
+			burnout_signal.data["sender"] = src.net_id
+			burnout_signal.data["address_1"] = src.linked_address
+			burnout_signal.data["command"] = "ping_reply"
+			burnout_signal.data["status"] = MARIONETTE_IMPLANT_STATUS_BURNED_OUT
+			SEND_SIGNAL(src, COMSIG_MOVABLE_POST_RADIO_PACKET, burnout_signal)
+		src.name = "melted [src.name]" // Specifically change the name here instead of using prefix, so that it appears in the removed implant item
+		src.desc = "Charred and most definitely broken. This thing must have been pushed really hard."
+		src.burned_out = TRUE
+		src.deactivate()
+
+
+/obj/item/remote/marionette_implant
+	name = "marionette implant remote"
+	desc = "A remote control that allows the sending and receiving of data from linked marionette implants."
+	icon = 'icons/obj/porters.dmi'
+	icon_state = "remote"
+	item_state = "electronic"
```

Comment:
This doesn't actually seem to have an inhand, but that might be for the best considering it's meant to be somewhat sneaky.